src/OpenTelemetry.Abstractions/Trace/ISpanBuilder.cs

```diff
@@ -53,6 +62,14 @@ public interface ISpanBuilder
         /// <returns>This span builder for chaining.</returns>
         ISpanBuilder SetNoParent();
 
+        /// <summary>
+        /// Set the <see cref="Activity.Current"/> to create span from. This span will represent the Activity,
+        /// not the child of it. Use it when library creates Activity on its own.
+        /// Any parent that was set previously will be discarded.
+        /// </summary>
+        /// <returns>This span builder for chaining.</returns>
+        ISpanBuilder FromCurrentActivity();
```

Comment:
this method ties `SpanBuilder` to the context. Which is very misaligned with API. Would it be better to have FromActivity method and user of API will pass `Activity.Current` explicitly?

Reply:
To be fair, SpanBuilder is tied to context already by using the current span unless any other parent was provided. Why do you think it is misaligned with the API?
 
I do not see a good scenario for FromActivity(activity) except when it is current. Moreover, I want to strictly prohibit this usage because it internally complicates things even more.

Reply:
you are right. It is tight to the current span in `StartSpan` method. What I don't like in this API is that it is not like any other builder methods which simply add properties. Maybe it will be more logical to create the entire `SpanBuilder` from the current Activity from a Tracer rather than have this method on `SpanBuilder`.

Reply:
Wait, we have spanbuilder to build spans, tracer does not do it and let's not abuse it.

This API simply adds property at the end of the day. It looks different than other APIs, this is right.

How about we change it to something like SetAutocollected (true) - I'll try to come up with more descriptive name, but basically this is a bool flag, that in absence of parent, creates span from current activity.

Reply:
Spent great deal of time thinking about naming and I still believe `FromCurrentActivity` is the most accurate one. 
Still, this is naming - internally this API is no different than others - tells what context to use to create Span. I can come up with different names `AssociateWithCurrentActivity`  / `AssignToCurrentActivity()`

If you have strong objections here is the alternative

- `SetCreateChild(true/false)`. It will not be compatible with any SetParent methods (as parent assumes there will be a child) and will only apply to the case when there is implicit context. 
  - true: will create span which is child of current activity. this is default and equivalent to not acalling method at all
  - false: will continue current Activity without creating child

So I'll implement `SetCreateChild`. Let me know what you think @SergeyKanzhelev.